src/util/file-filter.js

```diff
@@ -0,0 +1,119 @@
+/* @flow */
+import path from 'path';
+
+import minimatch from 'minimatch';
+
+import {createLogger} from './logger';
+
+const log = createLogger(__filename);
+
+// Use this function to mimic path.resolve without resolving to absolute path.
+export const normalizeResolve = (file: string): string => {
+  // normalize
+  file = path.normalize(file);
+
+  // trim trailing slash
+  if (path.parse(file).base && file.endsWith(path.sep)) {
+    return file.slice(0, -1);
+  }
+  return file;
+};
+
+// FileFilter types and implementation.
+
+export type FileFilterOptions = {|
+  filesToIgnore?: Array<string>,
+  ignoreFiles?: Array<string>,
+  sourceDir?: string,
+  artifactsDir?: string,
+|};
+
+/*
+ * Allows or ignores files.
+ */
+export class FileFilter {
+  filesToIgnore: Array<string>;
+  sourceDir: ?string;
+
+  constructor({
+    filesToIgnore = [
```

Comment:
If we change `ignoreFiles` to `customFilesToIgnore`, every calls to `createFileFilter` would also need to be changed to `createFileFilter({sourceDir, artifactsDir, customFilesToIgnore: ignoreFiles})`.

How about to change `filesToIgnore` to `defaultIgnorePatterns` and keep `ignoreFiles` so `createFileFilter` could be called nicely as `createFileFilter({sourceDir, artifactsDir, ignoreFiles})`?